collector/receiverhelper/receiverhelper.go

```diff
@@ -0,0 +1,67 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package receiverhelper // import "go.opentelemetry.io/ebpf-profiler/collector/receiverhelper"
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/xconsumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/xreceiver"
+
+	"go.opentelemetry.io/ebpf-profiler/collector/internal"
+	"go.opentelemetry.io/ebpf-profiler/internal/controller"
+)
+
+var (
+	// ErrInvalidConfig is returned when the config is invalid.
+	ErrInvalidConfig = errors.New("invalid config")
+)
+
+// BuildProfilesReceiver builds a profiles receiver.
+// It creates a controller with the provided config and options.
+func BuildProfilesReceiver(
+	_ context.Context,
+	rs receiver.Settings,
+	baseCfg component.Config,
+	nextConsumer xconsumer.Profiles,
+	opts ...internal.Option) (xreceiver.Profiles, error) {
+	cfg, ok := baseCfg.(*Config)
+	if !ok {
+		return nil, ErrInvalidConfig
+	}
+
+	controlerCfg := controller.Config{
+		ReporterInterval:       cfg.ReporterInterval,
+		MonitorInterval:        cfg.MonitorInterval,
+		SamplesPerSecond:       cfg.SamplesPerSecond,
+		ProbabilisticInterval:  cfg.ProbabilisticInterval,
+		ProbabilisticThreshold: cfg.ProbabilisticThreshold,
+		Tracers:                cfg.Tracers,
+		ClockSyncInterval:      cfg.ClockSyncInterval,
+		SendErrorFrames:        cfg.SendErrorFrames,
+		VerboseMode:            cfg.VerboseMode,
+		OffCPUThreshold:        cfg.OffCPUThreshold,
+		IncludeEnvVars:         cfg.IncludeEnvVars,
+	}
+	return internal.NewController(&controlerCfg, rs, nextConsumer, opts...)
+}
+
+// Config is the configuration for the profiles receiver.
+type Config struct {
```

Comment:
I don't think receiverhelper is the right location for this config.
Receiverhelper provides tools to build custom receivers relying on ebpf-profiler. This config should be in the `collector` package.